src/services/organizeImports.ts

```diff
@@ -83,7 +84,12 @@ namespace ts.OrganizeImports {
         }
     }
 
-    function removeUnusedImports(oldImports: readonly ImportDeclaration[], sourceFile: SourceFile, program: Program) {
+    function removeUnusedImports(oldImports: readonly ImportDeclaration[], sourceFile: SourceFile, program: Program, allowDestructiveCodeActions: boolean) {
+        // As a precaution, if there are *syntax* errors in the file, consider unused import detection to be destructive (GH #43051)
+        if (!allowDestructiveCodeActions && program.getSyntacticDiagnostics(sourceFile).length) {

```

Comment:
I think the consensus from the meeting we had was that we should hold off on considering syntax errors until we’ve  seen that other mitigations aren’t sufficient. But @jessetrinity can correct me if I’m wrong.

Reply:
That makes it easier for me 🤠